### PR TITLE
fix: Custom Endpoint URL crash [HEAD-536]

### PR DESCRIPTION
**Pull request by asaf92**
### Description

Custom endpoints validation in the settings page did not work correctly and led to crashes.

### CODE REVIEW

File: Snyk.VisualStudio.Extension.Shared/CLI/SnykCli.cs [hunk 1]
```diff
@@ -81,7 +81,7 @@ public string GetApiToken()
             }
             catch (InvalidTokenException e)
             {
-                Logger.Error(e, "Error on get api token via cli for settings");
+                Logger.Warning(e, "Error on get api token via cli for settings");
```
Comment (bastiandoetsch) [inline Snyk.VisualStudio.Extension.Shared/CLI/SnykCli.cs:84]: `Error getting...` would probably be better.

File: Snyk.VisualStudio.Extension.Shared/Settings/SnykGeneralOptionsDialogPage.cs [hunk 2]
```diff
@@ -93,6 +93,12 @@ public string CustomEndpoint
             get => this.customEndpoint;
             set
             {
+                if (!Uri.IsWellFormedUriString(value, UriKind.Absolute))
```
Comment (bastiandoetsch) [inline Snyk.VisualStudio.Extension.Shared/Settings/SnykGeneralOptionsDialogPage.cs:96]: Should we add the same endpoint logic as in IntelliJ? https://github.com/snyk/snyk-intellij-plugin/blob/c56af2012a2974f7f52172c0a02effbd2a334523/src/main/kotlin/io/snyk/plugin/Utils.kt#L151

```
fun URI.isSnykTenant() =
    isSnykDomain() && (host.startsWith("app.") || host == "snyk.io" || isDev()) && path.endsWith("/api")
```

And then, when used for the API, the URL is transformed to an API URL:
```
fun URI.toSnykAPIv1(): URI {
    val host = host
        .replaceFirst("app.", "api.")
        .replaceFirst("deeproxy.", "api.")
        .prefixIfNot("api.")

    return URI(scheme, host, "/v1/", null)
}
```

Comment (asaf92) [reply Snyk.VisualStudio.Extension.Shared/Settings/SnykGeneralOptionsDialogPage.cs:96]: It looks like there are already places doing this conversion (`GetAppCustomEndpoint` for example)

**Review by bastiandoetsch — APPROVED** [review]
Thanks. Looks good! 🎉

**Pull request merged**